Packs/FeedGithub/Integrations/FeedGithub/FeedGitHub.py

```diff
@@ -0,0 +1,514 @@
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any Demisto logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this gitHub Feed implementation, no special attributes defined
+    """
+
+    def __init__(self, base_url: str, verify: bool, proxy: bool, owner: str, repo: str, headers: dict):
+        base_url = urljoin(base_url, f"/repos/{owner}/{repo}/")
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+
+    def get_repo_base_sha(self) -> str:
+        """
+        Retrieves the SHA of the base commit of the repository.
+
+        This function fetches all commits of the repository and returns the SHA of the latest commit,
+        which represents the base commit of the repository.
+
+        Returns:
+            str: The SHA of the base commit of the repository.
+        """
+        all_commits = []
+        response = self._http_request("GET", full_url=f"{self._base_url}commits", resp_type="response")
+        while "next" in response.links:
+            data = response.json()
+            all_commits.extend(data)
+            response = self._http_request("GET", full_url=response.links["next"]["url"], resp_type="response")
+        return all_commits[-1]["sha"]
+
+    def get_files_between_commits(self, base: str, head: str) -> list:
+        """
+        Retrieves the list of files modified between two commits.
+
+        Args:
+            base (str): The SHA of the base commit.
+            head (str): The SHA of the head commit.
+
+        Returns:
+            list: A list of files modified between the specified base and head commits.
+        """
+        return self._http_request("GET", f"compare/{base}...{head}")["files"]
+
+
+def parsing_files_by_status(commits_files: list) -> list:
+    """
+    Parses files from a list of commit files based on their status.
+
+    Args:
+        commits_files (list): A list of dictionaries representing commit files.
+
+    Returns:
+        list: A list of URLs for files that are added or modified.
+    """
+    relevant_files: list[dict] = []
+    for file in commits_files:
+        if file.get("status") == "added" or file.get("status") == "modified":
+            relevant_files.append(file.get("raw_url"))
+    return relevant_files
+
+
+def get_files_token(client: Client, relevant_files: list[str]):
+    """
+    Retrieves content of relevant files based on specified extensions.
+
+    Args:
+        client (Client): An instance of the client used for HTTP requests.
+        relevant_files (list): A list of URLs for relevant files.
+
+    Returns:
+        list: A list of file contents fetched via HTTP requests.
+    """
+    extensions_to_fetch = argToList(demisto.params().get("extensions_to_fetch") or [])
```

Comment:
generally not good practice to call demisto.params() outside of main. Lets pass this information in as an argument